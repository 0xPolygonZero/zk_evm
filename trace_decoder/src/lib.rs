#![feature(linked_list_cursors)]
#![feature(trait_alias)]
#![feature(iter_array_chunks)]
// TODO: address these lints
#![allow(unused)]
#![allow(private_interfaces)]

mod compact;
pub mod decoding;
mod deserializers;
pub mod processed_block_trace;
pub mod trace_protocol;
pub mod types;
pub mod utils;
