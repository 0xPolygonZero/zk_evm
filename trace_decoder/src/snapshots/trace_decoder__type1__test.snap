---
source: trace_decoder/src/lib.rs
expression: instructions
---
NonEmpty {
    inner: [
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    3,
                    6,
                    0,
                    1,
                    4,
                    6,
                    2,
                    0,
                    9,
                    3,
                    11,
                    5,
                    9,
                    4,
                    5,
                    13,
                    1,
                    6,
                    7,
                    6,
                    13,
                    15,
                    0,
                    9,
                    3,
                    4,
                    4,
                    6,
                    7,
                    9,
                    0,
                    15,
                    13,
                    3,
                    1,
                    11,
                    2,
                    0,
                    14,
                    7,
                    11,
                    1,
                    2,
                    10,
                    2,
                    14,
                    8,
                    14,
                    5,
                    14,
                    0,
                    9,
                    13,
                    0,
                    6,
                    8,
                    1,
                    0,
                    9,
                    6,
                    1,
                    6,
                    11,
                ],
            },
            nonce: None,
            balance: Some(
                10000000000000000000000,
            ),
            has_code: false,
            has_storage: false,
        },
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    4,
                    6,
                    8,
                    2,
                    8,
                    8,
                    0,
                    5,
                    6,
                    3,
                    1,
                    0,
                    12,
                    8,
                    2,
                    10,
                    10,
                    4,
                    12,
                    0,
                    1,
                    10,
                    7,
                    14,
                    1,
                    2,
                    10,
                    1,
                    0,
                    15,
                    8,
                    1,
                    1,
                    1,
                    10,
                    0,
                    5,
                    6,
                    0,
                    14,
                    7,
                    2,
                    11,
                    7,
                    0,
                    0,
                    5,
                    5,
                    5,
                    4,
                    7,
                    9,
                    0,
                    3,
                    1,
                    11,
                    8,
                    6,
                    12,
                    3,
                    5,
                    7,
                    13,
                ],
            },
            nonce: None,
            balance: Some(
                1,
            ),
            has_code: false,
            has_storage: false,
        },
        Hash {
            raw_hash: [
                26,
                105,
                126,
                129,
                71,
                88,
                40,
                25,
                114,
                252,
                209,
                59,
                201,
                112,
                125,
                188,
                210,
                241,
                149,
                152,
                107,
                5,
                70,
                61,
                123,
                120,
                66,
                101,
                8,
                68,
                90,
                4,
            ],
        },
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    11,
                    7,
                    0,
                    14,
                    8,
                    0,
                    5,
                    3,
                    8,
                    10,
                    12,
                    13,
                    10,
                    11,
                    13,
                    6,
                    1,
                    3,
                    7,
                    3,
                    5,
                    3,
                    11,
                    0,
                    15,
                    9,
                    13,
                    8,
                    13,
                    1,
                    4,
                    9,
                    15,
                    4,
                    13,
                    11,
                    10,
                    9,
                    1,
                    14,
                    8,
                    11,
                    14,
                    2,
                    14,
                    7,
                    9,
                    4,
                    6,
                    14,
                    4,
                    0,
                    9,
                    11,
                    15,
                    13,
                    11,
                    14,
                    6,
                    8,
                    5,
                    11,
                    9,
                ],
            },
            nonce: None,
            balance: Some(
                1,
            ),
            has_code: false,
            has_storage: false,
        },
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    8,
                    9,
                    8,
                    0,
                    2,
                    13,
                    6,
                    14,
                    13,
                    1,
                    10,
                    2,
                    8,
                    11,
                    0,
                    4,
                    9,
                    14,
                    9,
                    13,
                    4,
                    15,
                    14,
                    5,
                    3,
                    3,
                    4,
                    12,
                    5,
                    9,
                    0,
                    2,
                    15,
                    13,
                    9,
                    11,
                    12,
                    0,
                    0,
                    12,
                    4,
                    2,
                    8,
                    2,
                    1,
                    12,
                    8,
                    2,
                    15,
                    8,
                    2,
                    14,
                    14,
                    2,
                    13,
                    10,
                    1,
                    0,
                    11,
                    14,
                    9,
                    0,
                    8,
                ],
            },
            nonce: None,
            balance: Some(
                1,
            ),
            has_code: false,
            has_storage: false,
        },
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    5,
                    6,
                    2,
                    7,
                    4,
                    10,
                    2,
                    7,
                    13,
                    13,
                    7,
                    5,
                    2,
                    4,
                    9,
                    5,
                    5,
                    4,
                    1,
                    7,
                    12,
                    1,
                    1,
                    14,
                    12,
                    13,
                    9,
                    1,
                    7,
                    2,
                    5,
                    1,
                    12,
                    12,
                    7,
                    12,
                    4,
                    12,
                    8,
                    3,
                    1,
                    0,
                    15,
                    4,
                    12,
                    7,
                    14,
                    4,
                    11,
                    13,
                    3,
                    12,
                    3,
                    0,
                    4,
                    13,
                    3,
                    13,
                    9,
                    10,
                    7,
                    9,
                ],
            },
            nonce: None,
            balance: Some(
                10000000000000000000000,
            ),
            has_code: false,
            has_storage: false,
        },
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    3,
                    10,
                    11,
                    0,
                    9,
                    7,
                    0,
                    11,
                    7,
                    3,
                    8,
                    9,
                    5,
                    11,
                    8,
                    12,
                    9,
                    9,
                    5,
                    9,
                    11,
                    10,
                    14,
                    6,
                    8,
                    5,
                    12,
                    3,
                    10,
                    1,
                    9,
                    15,
                    4,
                    5,
                    14,
                    11,
                    5,
                    10,
                    13,
                    8,
                    9,
                    13,
                    4,
                    2,
                    11,
                    5,
                    2,
                    10,
                    3,
                    4,
                    0,
                    14,
                    12,
                    4,
                    10,
                    12,
                    2,
                    0,
                    4,
                    13,
                    1,
                    9,
                ],
            },
            nonce: None,
            balance: Some(
                1,
            ),
            has_code: false,
            has_storage: false,
        },
        Branch {
            mask: 4128,
        },
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    8,
                    7,
                    6,
                    13,
                    10,
                    5,
                    1,
                    8,
                    10,
                    3,
                    9,
                    3,
                    13,
                    11,
                    13,
                    0,
                    6,
                    7,
                    13,
                    12,
                    7,
                    2,
                    10,
                    11,
                    15,
                    10,
                    0,
                    8,
                    13,
                    4,
                    7,
                    5,
                    14,
                    13,
                    6,
                    4,
                    4,
                    7,
                    15,
                    12,
                    10,
                    9,
                    6,
                    13,
                    9,
                    2,
                    14,
                    12,
                    3,
                    15,
                    9,
                    14,
                    7,
                    14,
                    11,
                    10,
                    5,
                    0,
                    3,
                    12,
                    10,
                    6,
                    1,
                ],
            },
            nonce: None,
            balance: Some(
                1,
            ),
            has_code: false,
            has_storage: false,
        },
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    5,
                    2,
                    6,
                    8,
                    8,
                    10,
                    8,
                    15,
                    9,
                    2,
                    6,
                    12,
                    8,
                    1,
                    6,
                    12,
                    10,
                    1,
                    14,
                    0,
                    7,
                    9,
                    0,
                    6,
                    7,
                    12,
                    10,
                    11,
                    10,
                    9,
                    4,
                    4,
                    15,
                    1,
                    5,
                    8,
                    14,
                    7,
                    6,
                    4,
                    8,
                    1,
                    7,
                    11,
                    8,
                    3,
                    15,
                    12,
                    4,
                    3,
                    5,
                    9,
                    4,
                    3,
                    7,
                    0,
                    12,
                    10,
                    9,
                    12,
                    15,
                    6,
                    2,
                ],
            },
            nonce: None,
            balance: Some(
                1,
            ),
            has_code: false,
            has_storage: false,
        },
        AccountLeaf {
            key: NonEmpty {
                inner: [
                    6,
                    9,
                    0,
                    11,
                    2,
                    3,
                    9,
                    11,
                    10,
                    3,
                    10,
                    10,
                    15,
                    9,
                    9,
                    3,
                    14,
                    4,
                    4,
                    3,
                    10,
                    14,
                    1,
                    4,
                    10,
                    14,
                    15,
                    15,
                    12,
                    4,
                    4,
                    12,
                    15,
                    8,
                    13,
                    9,
                    9,
                    3,
                    1,
                    10,
                    7,
                    9,
                    11,
                    10,
                    14,
                    13,
                    9,
                    15,
                    10,
                    1,
                    4,
                    1,
                    13,
                    0,
                    14,
                    4,
                    5,
                    0,
                    6,
                    14,
                    1,
                    3,
                    1,
                ],
            },
            nonce: None,
            balance: Some(
                1,
            ),
            has_code: false,
            has_storage: false,
        },
        Branch {
            mask: 25971,
        },
    ],
}
